service/history/api/describeworkflow/api.go

```diff
@@ -46,6 +46,21 @@ import (
 	"go.temporal.io/server/service/history/workflow"
 )
 
+func clonePayloadMap(source map[string]*commonpb.Payload) map[string]*commonpb.Payload {
+	target := make(map[string]*commonpb.Payload, len(source))
+	for k, v := range source {
+		metadata := make(map[string][]byte, len(v.GetMetadata()))
+		for mk, mv := range v.GetMetadata() {
+			metadata[mk] = mv
```

Comment:
I think we also want to copy byte slices because, even though another thread can't append to this or delete from it, they can modify its elements.

Reply:
This won't happen in practice, I don't think we mutate any payload metadata or data directly anywhere, only replace the entire slice.
In any case it's safe in terms of serialization and would mitigate the panic since the slice's length won't change.